lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -420,6 +420,63 @@ void SPIRVRegularizeLLVMBase::cleanupConversionToNonStdIntegers(Module *M) {
   }
 }
 
+/// Move SPIRV builtin global variables to local memory
+/// Example: __spirv_BuiltInWorkgroupId is the builtin GV
+/// @__spirv_BuiltInWorkgroupId = external dso_local local_unnamed_addr
+///                               addrspace(1) constant <3 x i64>, align 32
+/// define weak_odr dso_local spir_kernel void @foo() {
+/// ...
+/// %0 = load i64, ptr addrspace(1) @__spirv_BuiltInWorkgroupId, align 32
+/// ret void
+/// }
+/// Transformed to --->
+/// @__spirv_BuiltInWorkgroupId = external dso_local local_unnamed_addr
+///                               addrspace(1) constant <3 x i64>, align 32
+/// define weak_odr dso_local spir_kernel void @foo() {
+/// ...
+/// %1 = load <3 x i64>, ptr addrspace(1) @__spirv_BuiltInWorkgroupId, align 32
+/// %__local_SPIRV_Builtin* = alloca <3 x i64> addrspace(0), align 32
+/// store <3 x i64> %1, ptr addrspace(0) %__local_SPIRV_Builtin*, align 32
+/// // Replace uses of __spirv_BuiltInWorkgroupId with %__local_SPIRV_Builtin*
+/// ret void
+/// }
+void SPIRVRegularizeLLVMBase::copyBuiltinGVToLocalVar(Module *M) {
```

Comment:
There are also cases, when get_..._id() builtin is represented via function call (for OpenCL and SPIR-V friendly). Are these cases being handled?

Reply:
yes. I ran into a LIT test fail due to this very same issue and I resolved it.